[Core] Fix Controls missing from Intellisense

jsuarezruiz opened the pull request:
### Description of Change ###

We're setting the EditorBrowsable to never in `BaseTemplatedView` and the attribute is also applies to inherited classes (AvatarView, Expander, etc).

We could keep the attribute on `BaseTemplatedView`, and set `[EditorBrowsable(EditorBrowsableState.Always)]` on each inherited class. But this will _"force"_ to remember this with each new contribution inheriting from `BaseTemplatedView`.

So, this PR removes the attribute and applies changes to make `BaseTemplatedView` private and abstract.

### Bugs Fixed ###

- Fixes #658

### API Changes ###

Removed `[EditorBrowsable(EditorBrowsableState.Never)]` in:

- BaseTemplatedView
- ValueConverterExtension
- MultiValueConverterExtension
- ValidationBehavior
- BaseBehavior
- BaseCommand
- BaseAsyncValueCommand
- BaseAsyncCommand

Moved the previous classes to `Internals` namespace.

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [ ] Has samples (if omitted, state reason in description)
- [ ] Rebased on top of main at time of PR
- [ ] Changes adhere to coding standard
<!-- If at all possible, please update/add the documentation on the repo below. We would very much appreciate that. If you are unable to, please consider at least opening an issue on the repo below so we know that Docs still need to be adjusted/created. Thanks! <3 -->
- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

AndreiMisiukevich reviewed (changes requested):
Please fix the next classes as well:
- ValueConverterExtension
- MultiValueConverterExtension
- ValidationBehavior
- BaseBehavior
- BaseCommand
- BaseAsyncValueCommand
- BaseAsyncCommand

AndreiMisiukevich commented:
BTW, I have a proposal: what if we will move all "EditorBrowsable Never" classes to separate namespaces?

```Xamarin.CommunityToolkit.ObjectModel``` to ```Xamarin.CommunityToolkit.ObjectModel.Internals```
```Xamarin.CommunityToolkit.Behaviors``` to ```Xamarin.CommunityToolkit.Behaviors.Internals```

etc.

It will allow avoiding having them visible in xaml

AndreiMisiukevich commented:
BaseTemplatedView and others must be preserved as public (due to inheritance issue)

jsuarezruiz commented:
@AndreiMisiukevich Wow, that's quick review! I have completed the changes. Removed `[EditorBrowsable(EditorBrowsableState.Never)] `in:

- BaseTemplatedView
- ValueConverterExtension
- MultiValueConverterExtension
- ValidationBehavior
- BaseBehavior
- BaseCommand
- BaseAsyncValueCommand
- BaseAsyncCommand

Moved the previous classes to `Internals` namespace.

AndreiMisiukevich commented on src/CommunityToolkit/Xamarin.CommunityToolkit/Views/BadgeView/BadgeView.shared.cs:
```diff
@@ -1,366 +1,367 @@
-		void OnBadgeIndicatorContainerPropertyChanged(object sender, PropertyChangedEventArgs e)
-			=> UpdateBadgeViewPlacement(true);
-	}
+    [ContentProperty(nameof(Content))]
```
Hmm, looks like tabs were replaced by spaces for some reason

pictos replied:
The bot can fix that

AndreiMisiukevich commented on src/CommunityToolkit/Xamarin.CommunityToolkit/Views/BaseTemplatedView.shared.cs:
```diff
@@ -1,14 +1,12 @@
 		protected TControl Control { get; private set; }
 
-		public BaseTemplatedView()
+		internal BaseTemplatedView()
```
Maybe let's preserve it public (e.g. somebody decides to build own view. Not sure...)

AndreiMisiukevich commented on src/CommunityToolkit/Xamarin.CommunityToolkit/Views/TabView/WindowsTabViewItemTemplate.shared.cs:
```diff
@@ -5,7 +5,7 @@
 namespace Xamarin.CommunityToolkit.UI.Views
 {
 	[Preserve(AllMembers = true)]
```
Side note. Not related to this PR: I guess we should revisit all files and remove ```[Preserve(AllMembers = true)]```
Replace them by @pictos workaround ```if (DateTiem.Now.Ticks ....)```

pictos replied:
I can do it next week, if you want. Probably this will be a mouse  cat game

pictos reviewed (approved):
That LGTM. Thanks for this quick action!

jsuarezruiz commented:
Waiting the Build before merge!

jfversluis commented:
Build failed :( tests seem to fail

jsuarezruiz commented:
We made some changes in namespaces and we have tests validating namespaces. That's why a couple of tests failed. The build is ok now.
